model/client.go

```diff
@@ -291,6 +294,11 @@ func (c *Client) Logout() (*Result, *AppError) {
 	}
 }
 
+func (c *Client) SetOAuthToken(token string) {
+	c.AuthToken = token
+	c.AuthType = HEADER_TOKEN
```

Comment:
maybe rename HEADER_TOKEN to HEADER_OAUTH_TOKEN.  seems weird you're setting type to just token.